src/runtime/typemanager.cs

```diff
@@ -698,8 +698,21 @@ internal static void InitializeNativeCodePage()
             // to be executable.
             IMemoryMapper mapper = CreateMemoryMapper();
             int codeLength = NativeCode.Active.Code.Length;
-            NativeCodePage = mapper.MapWriteable(codeLength);
-            Marshal.Copy(NativeCode.Active.Code, 0, NativeCodePage, codeLength);
+
+            // Request 7 bytes more than required such that we can align to an 8 byte boundary on ARM
+            NativeCodePage = mapper.MapWriteable(codeLength + 7);
```

Comment:
mmap will return minimum a full page, and pages are typically 4096-byte-aligned (certainly more than 8-byte aligned). Add some logging to make sure, but I don't expect your changes are actually doing anything.

After the bx lr you might need a nop? I don't know this ISA myself, but it's not uncommon in architectures with branch prediction (i.e. basically all of them) that it'll unconditionally run an opcode right after a branch.

Reply:
https://community.arm.com/developer/ip-products/processors/f/cortex-m-forum/7154/alignment-in-arm

Looks like you need 16-byte-aligned if it's 64-bit ARM. So your Return1 is not properly aligned.

Reply:
The older RPis are 32bit, though, so this (8 byte alignment) can't be the current issue.